core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -241,13 +242,49 @@ class QualificationAppInfo(
     stages.map { stageId =>
       val stageTaskTime = stageIdToTaskEndSum.get(stageId)
         .map(_.totalTaskDuration).getOrElse(0L)
+      val transitionsTime = stageIdToGpuCpuTransitions.getOrElse(stageId, 0) match {
+        case gpuCpuTransitions if gpuCpuTransitions > 0 =>
+          // Duration to transfer data from GPU to CPU and vice versa.
+          // Assuming it's a PCI-E Gen3, but also assuming that some of the result could be
+          // spilled to disk.
+          // Duration in Spark metrics is in millisecond, so multiply this by 1000 to make
+          // it consistent
+          val totalBytesRead =
+            stageIdToTaskEndSum.get(stageId).map(_.totalbytesRead).getOrElse(0L).toDouble
+          if (totalBytesRead > 0) {
```

Comment:
so if we have transitions but bytes read is zero or small then we don't penalize at all?  just wondering how accurate that really is.  I think we should handle at least that case and penalize somehow

Reply:
Currently we penalize only those transitions where size is known. I will check if there is any way we could penalize.